docs/en/operations/storing-data.md

```diff
@@ -5,53 +5,404 @@ sidebar_label: "External Disks for Storing Data"
+- `s3_max_put_rps` — Maximum PUT requests per second rate before throttling. Default value is `0` (unlimited).
+- `s3_max_put_burst` — Max number of requests that can be issued simultaneously before hitting request per second limit. By default (`0` value) equals to `s3_max_put_rps`.
+- `s3_max_get_rps` — Maximum GET requests per second rate before throttling. Default value is `0` (unlimited).
+- `s3_max_get_burst` — Max number of requests that can be issued simultaneously before hitting request per second limit. By default (`0` value) equals to `s3_max_get_rps`.
+- `read_resource` — Resource name to be used for [scheduling](/docs/en/operations/workload-scheduling.md) of read requests to this disk. Default value is empty string (IO scheduling is not enabled for this disk).
+- `write_resource` — Resource name to be used for [scheduling](/docs/en/operations/workload-scheduling.md) of write requests to this disk. Default value is empty string (IO scheduling is not enabled for this disk).
+- `key_template` — Define the format with which the object keys are generated. By default, Clickhouse takes `root path` from `endpoint` option and adds random generated suffix. That suffix is a dir with 3 random symbols and a file name with 29 random symbols. With that option you have a full control how to the object keys are generated. Some usage scenarios require having random symbols in the prefix or in the middle of object key. For example: `[a-z]{3}-prefix-random/constant-part/random-middle-[a-z]{3}/random-suffix-[a-z]{29}`. The value is parsed with [`re2`](https://github.com/google/re2/wiki/Syntax). Only some subset of the syntax is supported. Check if your preferred format is supported before using that option. Disk isn't initialized if clickhouse is unable to generate a key by the value of `key_template`. It requires enabled feature flag [storage_metadata_write_full_object_key](/docs/en/operations/settings/settings#storage_metadata_write_full_object_key). It forbids declaring the `root path` in `endpoint` option. It requires definition of the option `key_compatibility_prefix`.
+- `key_compatibility_prefix` — That option is required when option `key_template` is in use. In order to be able to read the objects keys which were stored in the metadata files with the metadata version lower that `VERSION_FULL_OBJECT_KEY`, the previous `root path` from the `endpoint` option should be set here.
+
+:::note
+Google Cloud Storage (GCS) is also supported using the type `s3`. See [GCS backed MergeTree](/docs/en/integrations/gcs).
+:::
+
+### Using Plain Storage {#s3-storage}
+
+There is a disk type `s3_plain`, which provides a write-once storage. Unlike `s3` disk type, it stores data as is, e.g. instead of randomly-generated blob names, it uses normal file names as clickhouse stores files on local disk. So this disk type allows to keeper a static version of the table and can also be used to create backups on it.
```

Comment:
Shouldn't this talk about the `plain` metadata type now, since this setting no longer exists?

Reply:
Also, because it's write-once, does this mean it never merges? Or when it merges it never deletes old parts?

Reply:
> So this disk type allows to keeper a static version of the table and can also be used to create backups on it. Configuration parameters are the same as for s3 disk type.

Doesn't mean much to me tbh. Does this mean it basically just puts the exact clickhouse path on S3? If so, why are we doing the random blobs locally then, so metadata is then stored locally? Maybe that confusion is a result of this not being updated to reflect the `plain` metadata_type.

It would be useful to show this in context as well, like when I do this:
- what does the S3 file structure look like?
- how does this affect merges (as previously mentioned)?
- should this ever be a working-set table, or do I need to make this a materialized view target? If this is a working table, what are the performance implications?
- what do you mean by keep a static version and create a backup? Are you saying that because merges never delete it's a backup? Confusing without deep clickhouse context

Reply:
> Shouldn't this talk about the plain metadata type now, since this setting no longer exists?

Yes, this made me to remember that some code was missing to allow usage of plain metadata for other object storage types, which was addressed in https://github.com/ClickHouse/ClickHouse/pull/60396, so this is why I did not continue with this doc until now, needed to merge the PR first.

> Also, because it's write-once, does this mean it never merges? Or when it merges it never deletes old parts?

Yes, merges are disabled. And Inserts are not allowed as well (an exception will be thrown on an attempt to insert some data). Added this to doc.

> Doesn't mean much to me tbh.

Added some more explanation in doc.

> Does this mean it basically just puts the exact clickhouse path on S3?

Yes.

> If so, why are we doing the random blobs locally then, so metadata is then stored locally?

For `s3` disk type we store data in random blobs because unlike `s3_plain` it is not "write once", e.g. we have inserts and merges, so requirements are higher for ordinary s3. The limitations of object storage (no rename, move, hardlink operations, etc) do not allow the same usability as local filesystem allows, therefore we cannot handle it the same way.

> what does the S3 file structure look like?

Just randomly generated strings with 3-digit prefix, e.g. `/prefix_from_disk_config/blob_random_3_digit_prefix/blob_random_name`.
Also there was some feature which allows to change this blob path representation in a more performant way https://github.com/ClickHouse/ClickHouse/pull/57663, as I see it was documented already.

> should this ever be a working-set table, or do I need to make this a materialized view target?

This is a normal read-only table, you can do whatever you want. The initial use case for s3_plain disk was to create backups to it (I added some info to doc about it). Backups to any other disk type apart from plain is not allowed.

> If this is a working table, what are the performance implications?

None apart from data parts not being merged.

> what do you mean by keep a static version and create a backup?

Added explanation in doc.

Reply:
@danthegoodman1 I will merge this PR for now. If you have more comments please write - I will address them in the next PR.